psi4/src/psi4/libmints/potential.cc

```diff
@@ -48,826 +50,64 @@ using namespace psi;
 PotentialInt::PotentialInt(std::vector<SphericalTransform> &st, std::shared_ptr<BasisSet> bs1,
                            std::shared_ptr<BasisSet> bs2, int deriv)
     : OneBodyAOInt(st, bs1, bs2, deriv) {
-    if (deriv == 0)
-        potential_recur_ = new ObaraSaikaTwoCenterVIRecursion(bs1->max_am() + 1, bs2->max_am() + 1);
-    else if (deriv == 1)
-        potential_recur_ = new ObaraSaikaTwoCenterVIDerivRecursion(bs1->max_am() + 2, bs2->max_am() + 2);
-    else if (deriv == 2)
-        potential_recur_ = new ObaraSaikaTwoCenterVIDeriv2Recursion(bs1->max_am() + 3, bs2->max_am() + 3);
-    else
-        throw PSIEXCEPTION("PotentialInt: deriv > 2 is not supported.");
-
-    const int maxam1 = bs1_->max_am();
-    const int maxam2 = bs2_->max_am();
-
-    int maxnao1 = INT_NCART(maxam1);
-    int maxnao2 = INT_NCART(maxam2);
-
-    if (deriv == 1) {
-        // We set chunk count for normalize_am and pure_transform
-        // We can't use the trick of using less memory that I implemented in overlap & kinetic
-        // since potential integral derivatives also have a contribution to center c...which is
-        // over all atoms.
-        set_chunks(3 * natom_);
-
-        maxnao1 *= 3 * natom_;
-    } else if (deriv == 2) {
-        set_chunks(27 * natom_);
-        maxnao1 *= 27 * natom_;
+    if (bs1 != bs2) {
+        outfile->Printf("*********************************************************************************************************************\n");
+        outfile->Printf("When computing potential integrals with different bra and ket basis, the atom definition is taken from the bra basis.\n");
```

Comment:
What's "the atom definition"? Why does only this file have such a warning?

Reply:
Oh, because potential integrals are a weird corner case when thinking about 1e ints - they have 3 centers involved instead of 2.  The 3rd center corresponds to the nuclear charge of each center, which is extracted from the bra basis set's molecule.